File: 3cat-backend/app/schemas/employee.py
from pydantic import BaseModel, EmailStr
from typing import Optional
from datetime import datetime
from typing import List

class MatchEmployeesRequest(BaseModel):
    user_ids: List[str]


class EmployeeBase(BaseModel):
    email: Optional[EmailStr] = None
    bamboo_hr_id: Optional[str] = None
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    department: Optional[str] = None
    current_compensation: Optional[float] = None
    gusto_id: Optional[str] = None
    sevenshift_id: Optional[str] = None
    

class EmployeeCreate(EmployeeBase):
    email: EmailStr
    bamboo_hr_id: str
    first_name: str
    last_name: str
    location_id: Optional[int] = None
    department: Optional[str] = None
    current_compensation: Optional[float] = None
    gusto_id: Optional[str] = None
    sevenshift_id: Optional[str] = None
    current_title_id: Optional[int] = None
    current_level_id: Optional[int] = None
    department_id: Optional[int] = None
    punch_id: Optional[int] = None
    status: Optional[str] = None
   

class EmployeeUpdate(EmployeeBase):
    bamboo_hr_id: Optional[str] = None
    sevenshift_id: Optional[str] = None
    current_compensation: Optional[float] = None
    email: Optional[EmailStr] = None
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    department: Optional[str] = None
    current_title_id: Optional[int] = None
    current_level_id: Optional[int] = None
    location_id: Optional[int] = None
    department_id: Optional[int] = None
    pay_rate: Optional[float] = None
    punch_id: Optional[int] = None
    status: Optional[str] = None
    
    class Config:
        orm_mode = True

class Employee(EmployeeBase):
    id: int
    
    class Config:
        orm_mode = True



class EmployeeResponse(BaseModel):
    employee_id: int
    email: EmailStr
    bamboo_hr_id: Optional[str]
    first_name: Optional[str]
    last_name: Optional[str]
    department: Optional[str]
    created_at: Optional[datetime]
    updated_at: Optional[datetime]
    department_id: Optional[int]
    current_level_id: Optional[int]
    current_title_id: Optional[int]
    location_id: Optional[int]
    manager_id: Optional[int]
    phone: Optional[str]
    current_compensation: Optional[float]
    gusto_id: Optional[str]
    sevenshift_id: Optional[str]
    punch_id: Optional[int]

    class Config:
        orm_mode = True

File: 3cat-frontend/src/features/team/types/employee.ts
// src/types/employee.ts

// Base employee interface matching your database schema
export interface Employee {
  employee_id: number;
  email: string;
  bamboo_hr_id: string | null;
  first_name: string;
  last_name: string;
  department: string;
  created_at: string;
  updated_at: string;
  department_id: number;
  current_level_id: number;
  current_title_id: number;
  location_id: number;
  manager_id: number;
  phone: string | null;
  current_compensation: number | null;
  gusto_id: string | null;
  sevenshift_id: string | null;
  punch_id: string | null;
  status: string;
}

// Raw employee data from API (matching your current EmployeeData interface)
export interface EmployeeApiResponse {
  id: number;
  first_name: string;
  last_name: string;
  email: string;
  phone: string | null;
  current_title_id: string | null;
  location_id: number | null;
  status: string;
  bamboo_hr_id: string | null;
  sevenshift_id: string | null;
  gusto_id: string | null;
  punch_id: string | null;
  created_at: string;
  current_compensation: number | null;
  hourlyRate: number | null;
}

// Enhanced employee interface with UI display fields
export interface EmployeeWithUIFields extends EmployeeApiResponse {
  name: string;
  initials: string;
  avatar: string;
}

// Job Title interface
export interface JobTitle {
  title_id: number;
  title_name: string;
}

// Location interface
export interface Location {
  location_id: number;
  location_code: string;
  location_name: string;
}

// Employee creation interface
export interface EmployeeCreate {
  email: string;
  bamboo_hr_id?: string;
  first_name: string;
  last_name: string;
  department?: string;
  department_id?: number;
  current_level_id?: number;
  current_title_id?: number;
  location_id?: number;
  manager_id?: number;
  phone?: string;
  current_compensation?: number;
  gusto_id?: string;
  sevenshift_id?: string;
  punch_id?: string;
  status?: string;
}

// Employee update interface
export interface EmployeeUpdate {
  bamboo_hr_id?: string | null;
  sevenshift_id?: string | null;
  current_compensation?: number | null;
  email?: string;
  first_name?: string;
  last_name?: string;
  department?: string;
  current_title_id?: number | null;
  current_level_id?: number | null;
  location_id?: number | null;
  department_id?: number | null;
  pay_rate?: number | null;
  punch_id?: string | null;
  gusto_id?: string | null;
  status?: "Active" | "On Leave" | "Inactive" | "Part-Time" | "Full-Time" | "Terminated";
}